library/SimplePie/Locator.php

```diff
@@ -75,12 +75,19 @@ public function __construct(SimplePie_File $file, $timeout = 10, $useragent = nu
 		$this->force_fsockopen = $force_fsockopen;
 		$this->curl_options = $curl_options;
 
-		if (class_exists('DOMDocument'))
+		if (class_exists('DOMDocument') && $this->file->body != '')
 		{
 			$this->dom = new DOMDocument();
 
 			set_error_handler(array('SimplePie_Misc', 'silence_errors'));
-			$this->dom->loadHTML($this->file->body);
+			try
+			{
+				$this->dom->loadHTML($this->file->body);
+			}
+			catch (Throwable $ex)
```

Comment:
Is this needed or is it more better safe than sorry?

Reply:
There are potentially other errors than just empty input. For this specific bug, using either the null/empty test or the catch is sufficient. The `catch` is to more specifically extend the `set_error_handler`.